wdl-lint/src/rules/redundant_input_assignment.rs

```diff
@@ -0,0 +1,103 @@
+//! A lint rule for flagging redundant input assignments
+
+use std::fmt::Debug;
+
+use wdl_ast::Ast::V1;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Node::NameRef;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::v1::CallStatement;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Redundant Input Assignment rule.
+const ID: &str = "RedundantInputAssignment";
+
+/// Create a "Redundant Input Assignment" diagnostic.
+fn redundant_input_assignment(span: Span) -> Diagnostic {
+    Diagnostic::note("redundant input assignments can be shortened")
+        .with_rule(ID)
+        .with_label("redundant input assignments can be shortened", span)
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct RedundantInputAssignment(Option<SupportedVersion>);
+
+impl Rule for RedundantInputAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant input assignments."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Input assignments that are redundant can be shortened. For example, `{ input: a = a }` \
+         can be shortened to `{ input: a }`."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        None
```

Comment:
```suggestion
        Some(&[
            wdl_ast::SyntaxKind::VersionStatementNode,
            wdl_ast::SyntaxKind::WorkflowDefinitionNode,
            wdl_ast::SyntaxKind::CallStatementNode,
        ])
```
Your last rule was a special case and you didn't need to deal with exceptions, but this one needs to be "exceptable".

I would take a quick glance over my PR that introduced the current exception infrastructure if you are unclear how it works (#135).

Make sure you're test files check that `#@ except: RedundantInputAssignment` comments work as expected.